projects/element-ng/tour/si-tour.service.ts

```diff
@@ -179,17 +176,57 @@ export class SiTourService {
     }
   }
 
-  private makeOverlay(anchorElement: ElementRef<HTMLElement> | undefined): void {
-    const strategy = makePositionStrategy(anchorElement, this.overlay, 'auto');
-    this.handlePositionChangeSubscription(strategy, anchorElement);
+  private centerOverlay(): void {
+    this.positionChangeSub?.unsubscribe();
+    this.createOverlays();
+
+    const positionStrategy = this.overlayRef!.getConfig().positionStrategy;
+    if (!positionStrategy || positionStrategy instanceof FlexibleConnectedPositionStrategy) {
+      this.overlayRef!.updatePositionStrategy(
+        this.overlay.position().global().centerHorizontally().centerVertically()
+      );
+    }
+    this.tourToken.positionChange.next(undefined);
+  }
+
+  private attachOverlay(anchor: ElementRef<HTMLElement>): void {
+    this.createOverlays();
 
+    const positionStrategy = this.overlayRef!.getConfig().positionStrategy;
+    if (positionStrategy && positionStrategy instanceof FlexibleConnectedPositionStrategy) {
+      positionStrategy.setOrigin(anchor);
+    } else {
+      this.createFlexiblePositionStrategy(anchor);
+    }
+  }
+
+  private createFlexiblePositionStrategy(anchor: ElementRef<HTMLElement>): void {
+    const positionStrategy = this.overlay
+      .position()
+      .flexibleConnectedTo(anchor)
+      .withGrowAfterOpen(true)
+      .withPositions([
+        // On top
+        { originX: 'center', overlayX: 'center', originY: 'top', overlayY: 'bottom' },
+        { originX: 'start', overlayX: 'start', originY: 'top', overlayY: 'bottom' },
+        { originX: 'end', overlayX: 'end', originY: 'top', overlayY: 'bottom' },
+        // On bottom
+        { originX: 'center', overlayX: 'center', originY: 'bottom', overlayY: 'top' },
+        { originX: 'start', overlayX: 'start', originY: 'bottom', overlayY: 'top' },
+        { originX: 'end', overlayX: 'end', originY: 'bottom', overlayY: 'top' },
+        // Left and right
+        { originX: 'start', overlayX: 'end', originY: 'center', overlayY: 'center' },
+        { originX: 'end', overlayX: 'start', originY: 'center', overlayY: 'center' }
+      ]);
+    this.overlayRef!.updatePositionStrategy(positionStrategy);
+    this.positionChangeSub = positionStrategy.positionChanges
+      .pipe(map(change => ({ change, anchor })))
+      // We only want to forward the next channel, as the positionChanges completes when setting a new origin.
+      .subscribe(value => this.tourToken.positionChange.next(value));
+  }
+
+  private createOverlays(): void {
     if (this.overlayRef) {
-      this.overlayRef.updatePositionStrategy(strategy);
-      // This moves the dispatcher to the top, allowing it to catch other open overlays.
-      // Much lighter than to detach and re-attach the portal, not re-creating the si-tour
-      // component for each step
-      this.outsideClickDispatcher.remove(this.overlayRef);
-      this.outsideClickDispatcher.add(this.overlayRef);
```

Comment:
well, this is required and it does what the comment says. See the demo with the open context menu, now completely broken